src/ntfsck.c

```diff
@@ -2617,6 +2617,25 @@ static int ntfsck_scan_index_entries_btree(ntfs_volume *vol)
 			goto err_continue;
 		}
 
+		if (next->ie_flags == INDEX_ENTRY_END) {
```

Comment:
if next->ie_flags has both bits of INDEX_ENTRY_END and INDEX_ENTRY_NODE, then should ntfsck lookup sub-node?
It looks like to skip sub-node index, in this case.

Reply:
It only checks when it has the only index end flags bit(next->ie_flags == INDEX_ENTRY_END). If it have the index node bit , it will be checked below. Did I miss anything ?